Storage/Storage/CoreData/CoreDataManager.swift

```diff
@@ -213,6 +215,50 @@ public final class CoreDataManager: StorageManagerType {
             return debugMessages
         }
     }
+
+    /// Note that we have to enumerate all the files in our sandbox to get these properties, so this takes some time.
+    /// It is intended _only_ for use when logging a crash, or other high-value log which won't get in the user's way.
+    private static func storageSizeLogProperties() -> [String: Int64]? {
+        let directoryUrls = [
+            "Bundle": Bundle.main.bundleURL,
+            "Documents": FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first!,
+            "Library": FileManager.default.urls(for: .libraryDirectory, in: .userDomainMask).first!,
+            "Temp": FileManager.default.temporaryDirectory,
+        ]
+
+        let directorySizes = directoryUrls.mapValues { url in
+            getAllocatedStorageSize(for: url)
+        }
+
+        let totalSize = directorySizes.values.reduce(0) { partialResult, size in
+            return partialResult + size
+        }
+
+        return directorySizes.merging(["Total": totalSize]) { first, second in
+            first
+        }
+    }
+
+    /// Getting the size on disk isn't straightforward – this does not take into account all extended attributes, for example.
+    /// Generally the value returned here is an underestimate, but within 10% of the value iOS settings reports.
+    /// Using `fileAllocatedSize` as a fallback helps avoid counting some files as 0.
+    private static func getAllocatedStorageSize(for url: URL) -> Int64 {
+        let fileSizeKeys: [URLResourceKey] = [.totalFileAllocatedSizeKey, .fileAllocatedSizeKey]
+        guard let enumerator = FileManager.default.enumerator(at: url, includingPropertiesForKeys: fileSizeKeys) else {
+            return 0
+        }
+
+        var totalSize: Int64 = 0
+
+        for case let fileUrl as URL in enumerator {
+            if let fileSizeResource = try? fileUrl.resourceValues(forKeys: Set(fileSizeKeys)),
```

Comment:
nit;

`Set(fileSizeKeys)` could be theoretically moved up next to `fileSizeKeys` since we need the same at every iteration

Reply:
Yep, good call. I'm sure it's cheap, but we're doing it a lot for no reason.
e29c8d0